cmd/operator-controller/main.go

```diff
@@ -306,13 +306,14 @@ func main() {
 		os.Exit(1)
 	}
 
-	unpacker := &source.ContainersImageRegistry{
-		BaseCachePath: filepath.Join(cachePath, "unpack"),
-		SourceContextFunc: func(logger logr.Logger) (*types.SystemContext, error) {
+	imageCache := imageutil.BundleCache(filepath.Join(cachePath, "unpack"))
+	imagePuller := &imageutil.ContainersImagePuller{
+		SourceCtxFunc: func(ctx context.Context) (*types.SystemContext, error) {
 			srcContext := &types.SystemContext{
 				DockerCertPath: pullCasDir,
 				OCICertPath:    pullCasDir,
 			}
+			logger := logr.FromContextOrDiscard(ctx)
```

Comment:
Need to change this to be consistent with other `log.FromContext` (which we get from controller-runtime). This seems like a really easy thing to mess up. I wonder if we can implement another linter for this somehow.